coil-video/src/main/java/coil/fetch/MediaDataSourceFetcher.kt

```diff
@@ -0,0 +1,84 @@
+package coil.fetch
+
+import android.media.MediaDataSource
+import android.os.Build
+import androidx.annotation.RequiresApi
+import coil.ImageLoader
+import coil.annotation.ExperimentalCoilApi
+import coil.decode.DataSource
+import coil.decode.ImageSource
+import coil.request.Options
+import okio.Buffer
+import okio.Source
+import okio.Timeout
+import okio.buffer
+
+@RequiresApi(Build.VERSION_CODES.M)
+@OptIn(ExperimentalCoilApi::class)
+class MediaDataSourceFetcher(
+    private val data: MediaDataSource,
+    private val options: Options,
+) : Fetcher {
+
+    override suspend fun fetch(): FetchResult {
+        val imageSource = ImageSource(
+            source = MediaDataSourceOkIoSource(data).buffer(),
+            context = options.context,
+            metadata = MediaSourceMetadata(data),
+        )
+
+        return SourceResult(
+            source = imageSource,
+            mimeType = "video/", // TODO: How to get the mime type?
+            dataSource = DataSource.MEMORY, // TODO: MEMORY or DISK?
```

Comment:
I'd return `DataSource.DISK`. I tend to return the slowest data source it likely could be when the data source can't be determined.